app/models/download.rb

```diff
@@ -2,15 +2,31 @@ class Download
   COUNT_KEY     = "downloads".freeze
   ALL_KEY       = "downloads:all".freeze
 
-  def self.incr(name, full_name)
+  def self.incr(name, full_name, count: 1)
     today = Time.zone.today.to_s
-    Redis.current.incr(COUNT_KEY)
-    Redis.current.incr(rubygem_key(name))
-    Redis.current.incr(version_key(full_name))
-    Redis.current.zincrby(today_key, 1, full_name)
-    Redis.current.zincrby(ALL_KEY, 1, full_name)
-    Redis.current.hincrby(version_history_key(full_name), today, 1)
-    Redis.current.hincrby(rubygem_history_key(name), today, 1)
+    redis = Redis.current
+
+    redis.incrby(COUNT_KEY, count)
+    redis.incrby(rubygem_key(name), count)
+    redis.incrby(version_key(full_name), count)
+    redis.zincrby(today_key, count, full_name)
+    redis.zincrby(ALL_KEY, count, full_name)
+    redis.hincrby(version_history_key(full_name), today, count)
+    redis.hincrby(rubygem_history_key(name), today, count)
```

Comment:
@qrush:

> Could wrap all of these in a redis.pipelined block, it'll be a lot faster since we don't need to wait for any replies.

The `bulk_update` method that calls this already sets up a pipeline. :smile: